src/Mod/Sketcher/Gui/CommandSketcherTools.cpp

```diff
@@ -2522,3 +2529,27 @@ void CreateSketcherCommandsConstraintAccel()
     rcCmdMgr.addCommand(new CmdSketcherPaste());
 }
 // clang-format on
+
+void SketcherGui::centerScale(Sketcher::SketchObject* Obj, double scale_factor)
+{
+    std::vector<int> allGeoIds(Obj->Geometry.getValues().size());
+    std::iota(allGeoIds.begin(), allGeoIds.end(), 0);
+
+
+    auto doc = Gui::Application::Instance->activeDocument();
+    SketcherGui::ViewProviderSketch* vp =
+        static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    auto scaler = DrawSketchHandlerScale::make_centerScale(allGeoIds, scale_factor, false);
```

Comment:
`auto` should be used if: 
a. the right-hand side makes clear which type it is, e.g. with static_cast, or new 
b. the type is long or verbose (iterators, ranges, …) 
c. it reduces redundancy in code (repeated type)

In that case probably
```
    App::Document* doc = Gui::Application::Instance->activeDocument();
    auto* vp = static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
    auto scaler = DrawSketchHandlerScale::make_centerScale(allGeoIds, scale_factor, false);
```

`scaler` is in grey area - it is hard to deduce type but `std::unique_ptr<DrawSketchHandlerScale>` is pretty long one.